### ISSUE TITLE
Template not found error when conditionally including a dynamic partial

### ISSUE THREAD

**Issue by jmirmina** [linkedin/dustjs#730]
We're trying to upgrade to 2.7.2, and I am running into an error that is similar to the one in Issue #691:

```
Template Not Found: function body_4(chk,ctx){return chk.reference(ctx.get(["optionalPartial"], false),ctx,"h");}
```

Here is a simplified version of the template:

```
{?optionalPartial}{>"{optionalPartial}" /}{/optionalPartial}
```

Where optionalPartial will either be a path to another template or false. I tracked it down to something that changed between 2.6.1 and 2.6.2. I believe it has to do with the context that is being used to evaluate the partial name inside of the if block, because the following both work:

```
{?optionalPartial}{>"{optionalPartial}":context /}{/optionalPartial}
```
which results in the wrong context inside of the partial, and
```
{>"{optionalPartial}" /}
```

**Comment by sethkinast** [comment]
Do you use an alternate compiler like the guy in #691? It sounds like you
are using a compiler that passes the wrong context.

**Comment by jmirmina** [comment]
No, we are calling ```dust.compile``` to compile the templates.

**Comment by sethkinast** [comment]
Will dig in and take a look. Thanks for taking the time to narrow it down
to 2.6.2!

**Comment by sethkinast** [comment]
Here is the sample app I'm using to test for this.

```
var dust = require('dustjs-linkedin');

var tmpl = dust.compile('Dust {version}: {?optionalPartial}{>"{optionalPartial}" /}{/optionalPartial}', "hello");
var includeTmpl = dust.compile("I am included", "test");
dust.loadSource(tmpl);
dust.loadSource(includeTmpl);

dust.render("hello", { version: dust.version, optionalPartial: "test" }, function(err, out) {
  if(err) {
    console.error(err);
  } else {
    console.log(out);
  }
});
```

The output is `Dust 2.7.1: I am included` as I would expect. Are your templates perhaps compiled with an older version of the Dust compiler?

**Comment by jmirmina** [comment]
I ran your test and got the output ```Dust 2.7.2: I am included```, so it appears your example is working as expected in my environment. Ill dig in to this a bit and re-open this if I find anything else. I appreciate you taking the time to look into it.

**Comment by sethkinast** [comment]
I apologize that it took me so long; we've not had time as a team to dig into Dust for awhile. I will make sure that there is time for a followup if needed, just let me know.

Dust 2.7.0 broke compiler backwards compatibility but it was fixed in 2.7.1; maybe you had a copy of 2.7.0 kicking around?

**Comment by BennyH26** [comment]
Any updates on this?
